src/rx_signal_processing.py

```diff
@@ -81,22 +84,352 @@ def add_array(ndarray):
                 shm.close()
                 return name
 
-        main_corrs = data_outputs["main_corrs"][sd["slice_num"]]
-        output_dataset.main_acf_shm = add_array(main_corrs)
+        if cfs_scan_flag:
+            cfs_data = data_outputs["cfs_data"]
+            output_dataset.cfs_data = cfs_data[slice_num]
 
-        intf_available = True
-        try:
-            intf_corrs = data_outputs["intf_corrs"][sd["slice_num"]]
-            cross_corrs = data_outputs["cross_corrs"][sd["slice_num"]]
-        except KeyError as e:
-            # No interferometer data
-            intf_available = False
+            processed_data.add_output_dataset(output_dataset)
+            # if a clear frequency search was performed, add cfs data to message
+        else:
+            main_corrs = data_outputs["main_corrs"][sd["slice_num"]]
+            output_dataset.main_acf_shm = add_array(main_corrs)
 
-        if intf_available:
-            output_dataset.intf_acf_shm = add_array(intf_corrs)
-            output_dataset.xcf_shm = add_array(cross_corrs)
+            intf_available = True
+            try:
+                intf_corrs = data_outputs["intf_corrs"][sd["slice_num"]]
+                cross_corrs = data_outputs["cross_corrs"][sd["slice_num"]]
+            except KeyError as e:
+                # No interferometer data
+                intf_available = False
 
-        processed_data.add_output_dataset(output_dataset)
+            if intf_available:
+                output_dataset.intf_acf_shm = add_array(intf_corrs)
+                output_dataset.xcf_shm = add_array(cross_corrs)
+
+            processed_data.add_output_dataset(output_dataset)
+
+
+def sequence_worker(options, ringbuffer):
+
+    inproc_socket = zmq.Context().instance().socket(zmq.PAIR)
+    inproc_socket.connect("inproc://sqn_worker")
+
+    while True:
+        kwargs = inproc_socket.recv_pyobj()
+        # Wait until kwargs received from main thread
+
+        sequence_num = kwargs["sequence_num"]
+        main_beam_angles = kwargs["main_beam_angles"]
+        intf_beam_angles = kwargs["intf_beam_angles"]
+        mixing_freqs = kwargs["mixing_freqs"]
+        slice_details = kwargs["slice_details"]
+        start_sample = kwargs["start_sample"]
+        end_sample = kwargs["end_sample"]
+        processed_data = kwargs["processed_data"]
+        intf_antennas = kwargs["intf_antennas"]
+        filter_taps = kwargs["filter_taps"]
+        downsample_rates = kwargs["downsample_rates"]
+        cfs_scan_flag = kwargs["cfs_scan_flag"]
+        samples_needed = kwargs["samples_needed"]
+        rx_rate = kwargs["rx_rate"]
+        output_sample_rate = kwargs["output_sample_rate"]
```

Comment:
Might be nice if this was reformatted into a dataclass similar to `RadctrlParams`

Reply:
Created `RxProcessingParameters`